File: SRC/Manifest/Saras.eMarkingClient/src/app/model/project/setup/qig-management/qig-management-model.ts
export class QigManagementModel {
    QuestionCode!: string;
    QuestionMarks!: number;
    ProjectQigId!: number;
    QigName!: string;
    TolerenceLimit!: number;
    IsChildExist!: boolean;
    ParentQuestionId!: number;
    ProjectQuestionId!: number;
    QnsType!: number;
    QigType!: number;
    remarks!: string;
    IsSelected !: boolean;
    QuestionType!: number;
    IsSetupCompleted!: boolean;
    QigFibQuestions!: QigManagementModel[];
    TempQigFibQuestions!: QigManagementModel[];
}
export class ManageQigs {
    projectqigId!: number;
    QigName!: string;
    NoOfQuestions!: number;
    TotalMarks!: number;
    MarkingType!: string;
    QigType!: number;
    MandtoryQuestions!: number;
    IsQigSetupFinalized!: boolean;
    Remarks!: string;
    TotalNoOfQuestions?: number;
    TotalNoOfQIGs?: number;
    //public int TotalNoOfQuestions { get; set; }
    TotalNoOfTaggedQuestions?: number;
    TotalNoOfUnTaggedQuestions?: number;
}

export class ManageQigsCounts {
    TotalNoOfQIGs!: number;
    TotalNoOfQuestions!: number;
    TotalNoOfTaggedQuestions!: number;
    TotalNoOfUnTaggedQuestions!: number;
    IsProjectClosed!:number;
}
export class GetManagedQigListDetails {
    IsResetDisable!: boolean;
    IsResetEnable!:boolean;
    ManageQigsCountsList!: ManageQigsCounts;
    ManageQigsList!: ManageQigs[];
}

export class QigDetails {
    QigName!: string;
    NoOfQuestions!: number;
    TotalMarks!: number;
    MarkingType!: boolean;
    MandatoryQuestion?: number;
    IsQigSetup!: boolean;
    qigQuestions!: QigQuestions[];
}

export class QigQuestions {
    QigQuestionName!: string;
    TotalMarks!: number;
}

export class QigQuestionModel {
    QuestionCode!: string;
    QuestionType!: number;
    MaxMark!: number;
    QuestionText!: string;
    QuestionXML!: string;
}

export class QigQuestionsDetails {
    QuestionId!: number;
    QigQuestionName!: string;
    QigName!: string;
    QigTotalMarks!: number;
    TotalMarks!: number;
    QIGID!: number;
    QuestionType !: number;
    QigIds!: QignameDetails[];
}

export class QignameDetails {
    QIGID!: number;
    QIGName!: string;
    QIGCode!: string;
}

export class Tagqigdetails {
    ProjectQuestionId!: number;
    ProjectQigId!: number;
    MoveQigId!: number;
    QigTotalMarks!: number;
    QnsTotalMarks!: number;
}

export class CreateQigsModel {
    QigId!: number;
    QigName!: string;
    QigMarkingType!: number;
    QigMarkingTypeName!:string;
    ManadatoryQuestions!: number;
    QuestionMarkingType!: number;
    projectQuestions!: ProjectQuestionIds[];

}
export class FinalRemarks {
    Remarks!: string;
    projectqigId!: ProjectQuestionQigIds[];
}
export class ProjectQuestionQigIds {
    projectqigId?:number;
}
export class ProjectQuestionIds {
    ProjectQuestionId!: number;
    ParentQuestionId!: number;
}

export class BulkUserUploadModel {
    FirstName!: string;
    LastName!: string;
    LoginName!: string;
    RoleCode!: string;
    ESSNNumber!: string;
    PhoneNumber!: string;
}

export class LoginCredential {
  UserName!: string;
  Password!: string;
}

File: SRC/Service/Saras.eMarking/Saras.eMarking.Domain/ViewModels/Project/Setup/QigManagement/QigManagementModel.cs
using System;
using System.Collections.Generic;

namespace Saras.eMarking.Domain.ViewModels.Project.Setup.QigManagement
{
    public class QigManagementModel
    {
        public string QuestionCode { get; set; }
        public Decimal QuestionMarks { get; set; }
        public long ProjectQigId { get; set; }
        public string QigName { get; set; }
        public string remarks { get; set; }
        public int TolerenceLimit { get; set; }
        public Boolean IsChildExist { get; set; }
        public long? ParentQuestionId { get; set; }
        public long ProjectQuestionId { get; set; }
        public long QuestionType { get; set; }
        public Boolean IsSelected { get; set; }
        public Boolean IsSetupCompleted { get; set; }
        public Boolean IsComposite { get; set; }
        public int? QuestionOrder { get; set; }
        public List<QigManagementModel> QigFibQuestions { get; set; }

    }

    public class QigDetails
    {
        public long QigId { get; set; }
        public long ProjectId { get; set; }
        public string QigName { get; set; }
        public int NoOfQuestions { get; set; }
        public int TotalMarks { get; set; }
        public int MandatoryQuestion { get; set; }
        public string MarkingType { get; set; }
        public Boolean IsQigSetup { get; set; }
        public List<QigQuestions> qigQuestions { get; set; }
        public int? QigType { get; set; }

    }


    public class QigQuestions
    {
        public long QuestionId { get; set; }
        public string QigQuestionName { get; set; }
        public int TotalMarks { get; set; }
        public long ProjectQuestionId { get; set; }
        public int QuestionType { get; set; }
        public long ParentQuestionID { get; set; }
        public int? questionOrder { get; set; }
        public bool? IsComposite { get; set;}

    }
    public class QigQuestionsDetails
    {
        public long QuestionId { get; set; }
        public long QIGID { get; set; }
        public string QigQuestionName { get; set; }
        public string QigName { get; set; }
        public int? QuestionOrder { get; set; }

        public decimal? TotalMarks { get; set; }
        public decimal? QigTotalMarks { get; set; }
        public int QuestionType { get; set; }
        public List<QignameDetails> QigIds { get; set; }
    }

    public class QignameDetails
    {
        public long QIGID { get; set; }
        public string QIGName { get; set; }
        public string QIGCode { get; set; }
    }
    public class Tagqigdetails
    {
        public long ProjectQuestionId { get; set; }
        public long ProjectQigId { get; set; }
        public long MoveQigId { get; set; }
        public decimal? QnsTotalMarks { get; set; }
        public decimal? QigTotalMarks { get; set; }
    }

    public class GetManagedQigListDetails
    {
        public GetManagedQigListDetails()
        {
            ManageQigsCountsList = new ManageQigsCounts();
            ManageQigsList = new List<ManageQigs>();

        }
        public bool IsResetDisable { get; set; }
        public bool IsResetEnable { get; set; }
        public ManageQigsCounts ManageQigsCountsList { get; set; }
        public List<ManageQigs> ManageQigsList { get; set; }
    }

    public class ManageQigs
    {
        public long projectqigId { get; set; }
        public string QigName { get; set; }
        public int NoOfQuestions { get; set; }
        public Decimal TotalMarks { get; set; }
        public string MarkingType { get; set; }
        public int MandtoryQuestions { get; set; }
        public Boolean IsQigSetupFinalized { get; set; }

        public string Remarks { get; set; }
        public int? QigType { get; set; }
    }
    public class ManageQigsCounts
    {
        public int TotalNoOfQIGs { get; set; }
        public int TotalNoOfQuestions { get; set; }
        public int TotalNoOfTaggedQuestions { get; set; }
        public int TotalNoOfUnTaggedQuestions { get; set; }
        public int IsProjectClosed { get; set; }
    }

    public class CreateQigsModel
    {
        public string QigName { get; set; }
        public int MarkingType { get; set; }
        public byte? Qigtype { get; set; }
    }

    public class BlankQuestionDetails
    {
        public string QigName { get; set; }
        public long ProjectQuestionId { get; set; }
        public decimal? MaxScore { get; set; }
        public string QuestionCode { get; set; }

    }

    public class SaveQigQuestions
    {
        public long QigId { get; set; }
        public string QigName { get; set; }
        public int ManadatoryQuestions { get; set; }
        public int QigMarkingType { get; set; }
        public string QigMarkingTypeName { get; set; }
        public int MarkingType { get; set; }
        public List<ProjectQuestions> projectQuestions { get; set; }
    }

    public class ProjectQuestions
    {
        public long ProjectQuestionId { get; set; }
        public long ParentQuestionId { get; set; }
    }
    public class ProjectQuestionQigIds
    {
        public long ProjectQigId { get; set; }
    }
    public class FinalRemarks
    {
        public string remarks { get; set; }
        public List<ProjectQuestionQigIds> ProjectQigId { get; set; }

    }

    public class QigResetDetails
    {
        public string RoleCode { get; set; }
        public long ProjectUserRoleId { get; set; }
        public string Remarks { get; set; } 
        public long projectId { get; set; }

    }
}
